### ISSUE TITLE
Request: New Map feature to be added

### ISSUE THREAD

**Issue by viion** [ipinfo/php#30]
Hello!

I really love IP info and recently you guys launched: Map IPs: https://ipinfo.io/map

Would it be possible to get this in the library? I was thinking of PRing it but unsure if it's more of an "experimental/fun" thing at the moment or if it's something that will stick around and is production ready.

Kind Regards!

**Comment by UmanShahzad** [comment]
Hey @viion , really glad you like the new feature!

From my side, I don't see much value of putting this into the PHP SDK (or any SDK). The output is only going to be some JSON with a link like this:

```
$ cat ips.txt | curl -XPOST --data-binary @- "ipinfo.io/map?cli=1"
{
  "status": "Report Generated",
  "reportUrl": "https://ipinfo.io/map/7e55cd2d-a6d9-4724-bb6e-e0c8ac7daf6e"
}
```

If the intention is to input IPs to an SDK function and then use that link to be displayed somewhere in your app, then that would make sense. But I can't see any other use case. What do you think?

@coderholic any idea if this is worth putting into SDKs?

**Comment by viion** [comment]
I wonder if it's just be for convince, I guess it would align with your guys goals of this functionality.

**An example use case I have:** At my company we monitor AWS ALB logs, and we have reports generated daily about where traffic came from and this uses ipinfo php library to get region/city etc, and we build dashboards on that.

With the introduction of maps I have now added this to our report generation script:

```php
$ipfilename = self::SAVE_PATH .'/ip_list';
file_put_contents($ipfilename, implode("\n", $ips));

// generate map from ipinfo
$response = shell_exec('cat '. $ipfilename .' | curl -XPOST --data-binary @- "ipinfo.io/map?cli=1"');
$response = json_decode(trim($response), true);
$entity->setMapLink($response['reportUrl']);
```

So I'm basically just doing a `shell_exec` to run the example your page says, which works perfectly it's all fine and dandy, at the time I just thought "be handy if I could do `$ipinfo->generateMapLink($ips)` 😊"

Just thought i'd bring up to discuss, as ipinfo is so simple and convenient to use anyway!

**Comment by UmanShahzad** [comment]
I love that use case, thanks for sharing!

Will get back to you fairly soon about this; hoping we can add it :)

**Comment by UmanShahzad** [comment]
Alright got the 👍 to implement this; will do it soon.

**Comment by UmanShahzad** [comment]
Available in [v2.2.0](https://packagist.org/packages/ipinfo/ipinfo#v2.2.0).

Example from test cases:

```php
$h = new IPinfo();
$url = $h->getMapUrl(file("tests/map-ips.txt"));
echo "got URL ".$url;
```
